src/domain/occurrence/EditOccurrencePage.tsx

```diff
@@ -1,7 +1,155 @@
+import { Button } from 'hds-react';
 import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useParams } from 'react-router';
 
-const EditOccurrencePage = () => {
-  return <div>TODO: EDIT OCCURRENCE PAGE</div>;
+import BackButton from '../../common/components/backButton/BackButton';
+import LoadingSpinner from '../../common/components/loadingSpinner/LoadingSpinner';
+import {
+  useEditOccurrenceMutation,
+  useEventQuery,
+  useOccurrenceQuery,
+} from '../../generated/graphql';
+import useLocale from '../../hooks/useLocale';
+import formatDate from '../../utils/formatDate';
+import getLocalizedString from '../../utils/getLocalizedString';
+import scrollToTop from '../../utils/scrollToTop';
+import Container from '../app/layout/Container';
+import PageWrapper from '../app/layout/PageWrapper';
+import { ROUTES } from '../app/routes/constants';
+import EventOccurrenceForm, {
+  OccurrenceFormFields,
+} from './eventOccurrenceForm/EventOccurrenceForm';
+import styles from './occurrencePage.module.scss';
+import { getOccurrencePayload } from './utils';
+
+interface Params {
+  id: string;
+  occurrenceId: string;
+}
+
+const EditOccurrencePage: React.FC = () => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const history = useHistory();
+
+  const { id: eventId, occurrenceId } = useParams<Params>();
+
+  const { data: eventData, loading: loadingEvent } = useEventQuery({
+    variables: { id: eventId },
+  });
+
+  const [editOccurrence] = useEditOccurrenceMutation();
+
+  const {
+    data: occurrenceData,
+    loading: loadingOccurrence,
+  } = useOccurrenceQuery({
+    variables: { id: occurrenceId },
+  });
+
+  const goToEventDetailsPage = () => {
+    history.push(`/${locale}${ROUTES.EVENT_DETAILS.replace(':id', eventId)}`);
+  };
+
+  const goToOccurrencesPage = () => {
+    history.push(`/${locale}${ROUTES.OCCURRENCES.replace(':id', eventId)}`);
+  };
+
+  const getPayload = (values: OccurrenceFormFields) => {
+    return {
+      id: occurrenceId,
+      ...getOccurrencePayload(
```

Comment:
Consider passing arguments as an object and use destructuring in the function to avoid having to put so many arguments in the correct order